Library/Homebrew/brew.sh

```diff
@@ -26,8 +26,17 @@ esac
 # Higher depths mean this command was invoked by another Homebrew command.
 export HOMEBREW_COMMAND_DEPTH=$((HOMEBREW_COMMAND_DEPTH + 1))
 
+ohai() {
+  if [[ -t 1 && -z "$HOMEBREW_NO_COLOR" ]] # check whether stdout is a tty.
```

Comment:
Would be good to also use colours unconditionally if `HOMEBREW_COLOR` is set (but that can be a follow-up PR).